### PR TITLE
Build InvenioRDM docker images using GitHub Actions

### LINKED ISSUE #238: InvenioRDM Application Image
Using the Invenio base image with AlmaLinux, Python, and Node, and building a public InvenioRDM image that can be used in development, staging, and production. The initial version will be based on the InvenioRDM 10.0 release, and Docker images will be built with GitHub Actions and hosted in the GitHub Package Registry.

The basic building of the Docker image is straightforward, but we need to accommodate configuration options and define the relationship to `invenio-cli`. In an ideal world an Invenio-RDM dev instance should be started with `docker compose up` and a production instance via an Helm chart for Kubernetes.The Docker image should also allow deployment in production by other means, e.g ASWS ECS or Ansible.

### PR DESCRIPTION
**Pull request by mfenner**
### Description

Adding a GitHub Action that builds the InvenioRDM and Invenio-Nginx Docker images. They can be used to quickly start up InvenioRDM for local development (more work on SSL termination is needed before these images can be used in production). Using an AlmaLinux9 base image. Closes #238. 

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

### CODE REVIEW

File: {{cookiecutter.project_shortname}}/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,9 @@ FROM inveniosoftware/centos7-python:3.7
 {%- elif cookiecutter.python_version.split()[0] == '3.8' -%}
 FROM inveniosoftware/centos8-python:3.8
 {%- elif cookiecutter.python_version.split()[0] == '3.9' -%}
-FROM inveniosoftware/centos7-python:3.9
+# enabling multi-platform builds
+# using front-matter image until upstream image is available via inveniosoftware
+FROM --platform=$BUILDPLATFORM ghcr.io/front-matter/almalinux:1.0.0-beta.2
```
Comment (avivace) [inline {{cookiecutter.project_shortname}}/Dockerfile:18]: We should replace this with the image published on the CERN registry once https://github.com/inveniosoftware/docker-invenio/issues/51 is done

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3.1.0
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9' 
+      - name: Setup files and folders using cookiecutter
+        run: |
+          python -m pip install --upgrade pipenv cookiecutter
+          cookiecutter gh:front-matter/cookiecutter-invenio-rdm --no-input --config-file ./.github/default-config.yml
```
Comment (avivace) [inline .github/workflows/release.yml:28]: This remote should be changed to this repository's one

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ matrix.image }}
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
+          flavor: |
+            latest=true
+      - name: Set up Docker Buildx
```
Comment (avivace) [inline .github/workflows/release.yml:41]: @ntarocco Do we want such  images pushed to the CERN registry too?

**Review by avivace — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_shortname}}/Dockerfile (hunk 1 above)
Comment (mfenner) [reply {{cookiecutter.project_shortname}}/Dockerfile:18]: Agree

**Pull request closed without merging**